pcbnew2boardview.py

```diff
@@ -27,11 +27,15 @@ def y_coord(maxy, y, flipped):
     # Adjust y-coordinate to start from the bottom of the board and account for flipped components
     return coord(maxy - y) if not flipped else coord(y)
 
-def pad_sort_key(name):
-    if re.match(r"^\d+$", name):
-        return (0, int(name))
+def pad_sort_key(pad):
+    pad_name = pad.GetName()
+    
+    if pad_name == "":
+        return ('2',)
     else:
-        return (1, name)
+        parts = re.split('([0-9]+)', pad_name)
+        alphanumeric_parts = [f"{int(text):09d}" if text.isdigit() else text for text in parts if text]
```

Comment:
I think this will just as well, without hardcoding numbers:

```suggestion
        alphanumeric_parts = [text[::-1] if text.isdigit() else text for text in parts if text]
```

(It reverses the digit order by using a weird bit of Python syntax. It's basically `str(reversed(text))` but in one operation.)

Reply:
I just tested it and it still works great!

Reply:
I apologize, I forgot to hit "Refresh Plugins",  so even though I put your code into my plugins path, I was still running the previous code. Your solution results in this.
![image](https://github.com/whitequark/kicad-boardview/assets/53708281/3b439987-4ce3-4491-9438-8e1d294d9525)

Reply:
I think I may need to open a new PR but I will wait on confirmation from you to see if this can be reopened on your end.

I apologize for all of the disorder caused by my PRs.

Reply:
No this PR cannot be reopened since it was merged.

I will encourage you to learn how to use Git.

Reply:
I agree, I'm very sorry for the mix-up. I'm much better at hardware than I am software. I will make a new PR for this just so that you don't have to be reliant on the `.obdata` one to fix this issue.

Reply:
Also yeah there's a logic error, I think this should actually work:

```
import re # at the beginning of file

def natural_sort_key(s):
    return [int(text) if text.isdigit() else text.casefold()
            for text in re.compile('([0-9]+)').split(s)]
```